### ISSUE TITLE
forward rolling functions

### ISSUE THREAD

**Issue by hayd** [pandas-dev/pandas#6772]
https://groups.google.com/forum/#!msg/pydata/LLPnzMfDlSg/OpK-mHJzJwYJ
http://stackoverflow.com/questions/22820292/how-to-use-pandas-rolling-functions-on-a-forward-looking-basis

Applying to reverse Series and reversing could work on all (?) rolling functions, I think sometimes can just do on values array, a kwarg would be nice.

```
pd.rolling_(s[::-1])[::-1]
pd.Series(pd.rolling_(s.values[::-1])[::-1), s.index, name=s.name)  # faster?
```

(OP says that this doesn't actually work in some cases: http://stackoverflow.com/questions/22820292/how-to-use-pandas-rolling-functions-on-a-forward-looking-basis/22820689?noredirect=1#comment34813442_22820689... ?)

**Comment by luoshao23** [comment]
Is anybody still working on it ?

**Comment by ErfPy** [comment]
This is one I actually miss quite often. With datetime index you can't just flip reverse the dataframe, since you get the error: `ValueError: index must be monotonic`

**Comment by smeana** [comment]
If you have data for a fix interval, let's say every second, you can do shifting and then Rolling
shifting(-30) -> 30 seconds back and then rolling to find the mean, max, min on that window.

`df_rol['min_'+ str(t)] = df_new['FirstPrice'].shift(-1,freq=dt).rolling(dt).min()`

If you don't have a fix interval, I only found for looping as a solution after tried everything.
Working on billions of rows Looping approach is not possible.

**Comment by oldrichsmejkal** [comment]
Should be resolved by: https://github.com/pandas-dev/pandas/pull/28297

**Comment by mroeschke** [comment]
Yup looks like #28297 addresses this

**Comment by jfaleiro** [comment]
This issue should not be closed since #28297 (what seems to address this bug) isn't closed.

**Comment by benedekpasztor** [comment]
A possible solution is to shift the rolling window aggregation
